lib/src/merge.rs

```diff
@@ -501,6 +501,13 @@ impl MergedTreeValue {
                 .all(|value| matches!(value, Some(TreeValue::Tree(_)) | None))
     }
 
+    /// Whether this merge would resolve to an executable file
+    pub fn is_executable(&self) -> bool {
+        self.values
+            .iter()
+            .all(|v| matches!(*v, Some(TreeValue::File { executable, .. }) if executable))
+    }
```

Comment:
handle `is_present()` and `None` like `.is_tree()`?

Absent values should be excluded as we do in `to_file_merge()`. I'm not sure if an empty merge should be considered an executable, though.

Reply:
Alright, I'll change that (or maybe look for a version with slightly less unwrap ;_;).

Reply:
ok, i've got it using a merge of the executable bits now, and we only consider the conflict treevalue executable if that merge resolves